openedx_events/bridge/avro_attrs_bridge.py

```diff
@@ -1,253 +1,231 @@
-    }
-    # default config, should be overwritten by passing in config during obj initialization
-    default_config = {
-        "source": "/openedx/unknown/avro_attrs_bridge",
-        "sourcehost": "unknown",
-        "type": "org.openedx.test.test.test.v0",
+        DatetimeAvroAttrsBridgeExtension.cls: DatetimeAvroAttrsBridgeExtension(),
+        CourseKeyAvroAttrsBridgeExtension.cls: CourseKeyAvroAttrsBridgeExtension(),
     }
 
-    def __init__(self, attrs_cls, extensions=None, config=None):
+    def __init__(self, signal, extensions=None):
         """
         Init method for Avro Attrs Bridge.
 
         Arguments:
-            attrs_cls: Attr Class Object (not instance)
-            extensions: dict mapping Class Object to its AvroAttrsBridgeExtention subclass instance
-            config: dict with followings keys
-                - source:  This field will be used to indicate the logical source of an event, and will be of the form
-                           /{namespace}/{service}/{web|worker}.
-                           All services in standard distribution of Open edX should use openedx for the namespace.
-                           Examples of services might be “discovery”, “lms”, “studio”, etc.
-                           The value “web” will be used for events emitted by the web application,
-                           and “worker” will be used for events emitted by asynchronous tasks such as celery workers.
-                           For more info, see OEP-41: Asynchronous Server Event Message Format
-                - sourcehost: should represent the physical source of message
-                               -- i.e. host identifier of the server that emitted this event (example: edx.devstack.lms)
-                              For more info, see OEP-41: Asynchronous Server Event Message Format
-                - type: The name of event.
-                        Should be formatted `{Reverse DNS}.{Architecture Subdomain}.{Subject}.{Action}.{Major Version}`.
-                        For more info, see OEP-41: Asynchronous Server Event Message Format
+            signal: An instance of OpenEdxPublicSignal
+            extensions: dict mapping a Class to an instance of the required AvroAttrsBridgeExtension subclass, eg
+                       { MyDataClass : MyDataClassAvroAttrsBridgeExtension() }
         """
-        self._attrs_cls = attrs_cls
-
         self.extensions = {}
         self.extensions.update(self.default_extensions)
         if isinstance(extensions, dict):
             self.extensions.update(extensions)
 
-        self.config = {}
-        self.config.update(self.default_config)
-        if isinstance(config, dict):
-            self.config.update(config)
-
-        # Used by record_field_for_attrs_class function to keep track of which
+        self._signal = signal
+        # Used by _create_avro_field_definition function to keep track of which
         # records have already been defined in schema.
```

Comment:
The following potential comment would be more resilient to renames:
```suggestion
        # Used to track which records have already been defined in the schema.
```